packages/frontend/src/helpers/variables.ts

```diff
@@ -43,6 +47,36 @@ export interface Variable {
   isHidden?: boolean
 }
 
+function getStepName(executionStep: IExecutionStep, allApps: IApp[]) {
+  const { appKey, key, step } = executionStep
+
+  const isTrigger = ['formsg', 'webhook', 'scheduler'].includes(appKey)
+  const isIfThen = appKey === TOOLBOX_APP_KEY && key === TOOLBOX_ACTIONS.IfThen
+  const app = allApps?.find((a: IApp) => a.key === appKey)
+  const actionsOrTriggers: Array<ITrigger | IAction> =
+    (isTrigger ? app?.triggers : app?.actions) || []
+
+  const selectedActionOrTrigger = actionsOrTriggers.find(
+    (actionOrTrigger: IAction | ITrigger) => actionOrTrigger.key === key,
```

Comment:
key is not populated for old executions steps, so this is not really backwards compatible?